src/WebJobs.Script.WebHost/Diagnostics/FunctionsVersionLoggerService.cs

```diff
@@ -0,0 +1,128 @@
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _interval = TimeSpan.FromSeconds(this.Interval);
+            _logger = logger;
+        }
+
+        public virtual double Interval
+        {
+            get { return (_environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteSlotName) == "development") ? 3 : intervalInSecondsProd; }
+            set { intervalInSecondsProd = value; }
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            // start the timer by setting the due time
+            SetTimerInterval((int)_interval.TotalMilliseconds);
+
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            // stop the timer if it has been started
+            _timer?.Change(Timeout.Infinite, Timeout.Infinite);
+
+            return Task.CompletedTask;
+        }
+
+        private async void OnTimer(object state)
+        {
+            await PublishLogsSamplesAsync();
+        }
+
+        private Task PublishLogs()
+        {
+            try
+            {
+                if (_environment.IsKubernetesManagedHosting())
+                {
+                    _logger.LogInformation("FunctionsExtensionVersion : {0}", _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsExtensionVersion));
+                    _logger.LogInformation("Framework : {0}", _environment.GetEnvironmentVariable(EnvironmentSettingNames.Framework));
+                    _logger.LogInformation("FrameworkVersion : {0}", _environment.GetEnvironmentVariable(EnvironmentSettingNames.FrameworkVersion));
+                    _logger.LogInformation("SlotName : {0}", _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteSlotName));
+                }
+                return Task.CompletedTask;
+            }
+            catch (Exception exc) when (!exc.IsFatal())
+            {
+                return null;
+            }
+        }
+
+        private async Task PublishLogsSamplesAsync()
+        {
+            try
+            {
+                await PublishLogs();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning(ex, "Failed to publish function version logs");
+            }
+        }
+
+        private void SetTimerInterval(int dueTime)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+            var timer = _timer;
+            if (timer != null)
+            {
+                try
+                {
+                    _timer.Change(dueTime, Timeout.Infinite);
+                }
+                catch (ObjectDisposedException)
+                {
+                    // might race with dispose
+                    _logger.LogWarning("Exception in SetTimerInterval");
```

Comment:
Could other exceptions occur here or just `ObjectDisposedException`? Wondering if we need to account for a more generic Exception or not?

Also please have a clearly exception message here, i.e. something about the timer being disposed so we are unable to set the timer interval

Reply:
Done